js/ez-field-edit.js

```diff
@@ -0,0 +1,94 @@
+(function () {
+    /**
+    *  `<ez-field-edit>` is supposed to wrap one or several HTML5 inputs. It tracks user input,
+    *  and if an input is invalid, sets its own `invalid` property to true. It will also add an
+    * `ez-invalid` class on the related label. This class will be removed once the input gets valid.
+    * @polymerElement
+    * @demo demo/ez-field-edit.html
+    */
+    class FieldEdit extends Polymer.Element {
+        static get is() {
+            return 'ez-field-edit';
+        }
+
+        static get properties() {
+            return {
+                'invalid': {
+                    type: Boolean,
+                    value: false,
+                    reflectToAttribute: true,
+                },
+            };
+        }
+
+        constructor() {
+            super();
+            this.addEventListener('input', this._checkValidity.bind(this));
+        }
+
+        /**
+         * Checks whether the inputs are valid or not and set the `invalid` property
+         * accordingly.
+         * Also set `ez-invalid` class on labels connected to invalid inputs.
+         */
+        _checkValidity() {
+            let valid;
+
+            this.querySelectorAll('input, select, textarea').forEach((formElement) => {
+                valid = formElement.validity.valid;
+
+                this.querySelectorAll('label').forEach((labelElement) => {
+                    if (!valid) {
+                        this._commonAncestor(labelElement, formElement).classList.add('ez-invalid');
```

Comment:
you should put the invalid class in a const to avoid repeating it and also, I'm not super fan of that class, it's maybe a tad too generic. `ez-invalid-input` maybe ?